File: io/lucamienert/basic/ast/ImmediateString.java
package io.lucamienert.basic.ast;

import io.lucamienert.basic.instruction.Instruction;
import io.lucamienert.basic.instruction.InstructionSequence;
import io.lucamienert.basic.instruction.OpCode;

public class ImmediateString extends StringExpression {

    private final String value;

    public ImmediateString(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public void compile(InstructionSequence seq) {
        seq.append(new Instruction(OpCode.PUSHS, value));
    }

}

File: BasicCompiler/Ast/ImmediateString.cs
using BasicCompiler.Instructions;

namespace BasicCompiler.Ast;

public class ImmediateString : StringExpression
{
    public string Value { get; set; }

    public ImmediateString(string value)
        => Value = value;

    public override void Compile(InstructionSequence instructionSequence)
    {
        instructionSequence.Append(new Instruction(OpCode.PUSHS, Value));
    }
}
